akka-docs-dev/rst/java/code/docs/http/javadsl/HttpClientExampleDocTest.java

```diff
@@ -69,4 +73,25 @@ public void testSingleRequestExample() {
           .singleRequest(HttpRequest.create("http://akka.io"), materializer);
     //#single-request-example
   }
-}
\ No newline at end of file
+
+  static
+      //#single-request-in-actor-example
+    class Myself extends AbstractActor {
+      final Http http = Http.get(context().system());
+      final ExecutionContextExecutor dispatcher = context().dispatcher();
+      final ActorMaterializer materializer = ActorMaterializer.create(context().system());
+
+      public Myself() {
+        receive(ReceiveBuilder
+         .match(String.class, url -> {
+           Patterns.<HttpResponse>pipe(fetch(url), dispatcher).to(self());
```

Comment:
Hah, intellij can't figure it out though `javac` can.
It infers properly, removed the explicit type.

// feels like early years of scala when you couldn't trust your IDE, now the same happens to java... :wink: